tests/integration/test_fim/test_files/test_basic_usage/test_basic_usage_access_opened_files.py

```diff
@@ -0,0 +1,223 @@
+    - Ubuntu Focal
+    - Ubuntu Bionic
+    - Ubuntu Xenial
+    - Ubuntu Trusty
+    - Debian Buster
+    - Debian Stretch
+    - Debian Jessie
+    - Debian Wheezy
+    - Red Hat 8
+    - Red Hat 7
+    - Red Hat 6
+    - Windows 10
+
+references:
+    - https://documentation.wazuh.com/current/user-manual/capabilities/file-integrity/fim-configuration.html
+    - https://documentation.wazuh.com/current/user-manual/reference/ossec-conf/localfile.html
+
+tags:
+    - syscheck
+"""
+
+import os
+import random
+import string
+import pytest
+import time
+
+from wazuh_testing.fim import (
+    LOG_FILE_PATH,
+    generate_params,
+    detect_initial_scan_start,
+    get_scan_timestamp,
+)
+from wazuh_testing.tools import PREFIX
+from wazuh_testing.tools.configuration import (
+    load_wazuh_configurations,
+    check_apply_test,
+)
+from wazuh_testing.tools.monitoring import FileMonitor
+from wazuh_testing.tools.file import delete_path_recursively
+
+
+# Marks
+pytestmark = [pytest.mark.tier(level=0)]
+
+# Variables
+directory_str = os.path.join(PREFIX, "testdir1")
+file_path = os.path.join(directory_str, "large_file")
+wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+test_data_path = os.path.join(os.path.dirname(os.path.realpath(__file__)), "data")
+configurations_path = os.path.join(test_data_path, "wazuh_conf.yaml")
+
+# configurations
+conf_params = {"TEST_DIRECTORIES": directory_str, "MODULE_NAME": __name__}
+parameters, metadata = generate_params(extra_params=conf_params)
+configurations = load_wazuh_configurations(
+    configurations_path, __name__, params=parameters, metadata=metadata
+)
+
+
+# Fixtures
+@pytest.fixture(scope="module", params=configurations)
+def get_configuration(request):
+    """Get configurations from the module."""
+    return request.param
+
+
+@pytest.fixture(scope="function")
+def create_and_delete_large_file(get_configuration, request):
+    """
+    Create a large file and later delete the path.
+    """
+    if get_configuration["metadata"]["fim_mode"] != "scheduled":
+        pytest.skip("This test does not apply to realtime or whodata modes")
+    # If path exists delete it
+    if os.path.exists(directory_str):
+        delete_path_recursively(directory_str)
+    # create directory
+    os.mkdir(directory_str)
+    file_size = 1024 * 1024 * 960  # 968 MB
+    chunksize = 1024 * 768
+    # create file and write to it.
+    with open(file_path, "a") as f:
+        while os.stat(file_path).st_size < file_size:
+            f.write(random.choice(string.printable) * chunksize)
+    yield
+    # delete the file and path
+    delete_path_recursively(directory_str)
+
+
+@pytest.fixture(scope="function")
+def wait_for_scan_start(get_configuration, request):
+    """
+    Wait for start of initial FIM scan.
+    """
+    file_monitor = getattr(request.module, "wazuh_log_monitor")
+    try:
+        detect_initial_scan_start(file_monitor)
+    except KeyError:
+        detect_initial_scan_start(file_monitor)
+
```

Comment:
This function should be inside `fim.py` which has all the `callbacks`.
Also, if the functions already exist, **reuse** them, and if it is not considered suitable, **clarify why**.

Finally, you can see that both the `Try` and the `Except` call the same method. Is this correct?

Reply:
Moved over to test_fim/conftest.py with other fixtures.
It is different to wait_for_fim_start in that this fixture waits for the scheduled scan to start, while the other waits for it to end.